instrumentation/github.com/gin-gonic/gin/otelgin/test/gin_test.go

```diff
@@ -461,11 +461,25 @@ func TestMetrics(t *testing.T) {
 		name                        string
 		metricAttributeExtractor    func(*http.Request) []attribute.KeyValue
 		ginMetricAttributeExtractor func(*gin.Context) []attribute.KeyValue
+		requestTarget               string
+		expectHasRouteAttr          bool
+		expectStatus                int64
 	}{
 		{
 			name:                        "default",
 			metricAttributeExtractor:    nil,
 			ginMetricAttributeExtractor: nil,
+			requestTarget:               "/user/123",
+			expectHasRouteAttr:          true,
```

Comment:
This attribute isn't really extensible. Rather than a control flag, how about having a string `wantRouteAttr: "/user/:id"`?
If it's empty, we consider it shouldn't be set.

Then, future tests can have different values.